// @ionic-native/cloud-settings/__ivy_ngcc__/ngx/index.js
import { __decorate, __extends } from "tslib";
import { Injectable } from '@angular/core';
import { IonicNativePlugin, cordova } from '@ionic-native/core';
import * as ɵngcc0 from '@angular/core';
var CloudSettings = /** @class */ (function (_super) {
    __extends(CloudSettings, _super);
    function CloudSettings() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    CloudSettings.prototype.exists = function () { return cordova(this, "exists", {}, arguments); };
    CloudSettings.prototype.save = function (settings, overwrite) { return cordova(this, "save", { "successIndex": 1, "errorIndex": 2 }, arguments); };
    CloudSettings.prototype.load = function () { return cordova(this, "load", {}, arguments); };
    CloudSettings.prototype.onRestore = function (handler) { return cordova(this, "onRestore", { "sync": true }, arguments); };
    CloudSettings.prototype.enableDebug = function () { return cordova(this, "enableDebug", {}, arguments); };
    CloudSettings.pluginName = "CloudSettings";
    CloudSettings.plugin = "cordova-plugin-cloud-settings";
    CloudSettings.pluginRef = "cordova.plugin.cloudsettings";
    CloudSettings.repo = "https://github.com/dpa99c/cordova-plugin-cloud-settings";
    CloudSettings.install = "ionic cordova plugin add cordova-plugin-cloud-settings --variable ANDROID_BACKUP_SERVICE_KEY=myapikey";
    CloudSettings.installVariables = ["ANDROID_BACKUP_SERVICE_KEY"];
    CloudSettings.platforms = ["Android", "iOS"];
CloudSettings.ɵfac = function CloudSettings_Factory(t) { return ɵCloudSettings_BaseFactory(t || CloudSettings); };
CloudSettings.ɵprov = ɵngcc0.ɵɵdefineInjectable({ token: CloudSettings, factory: function (t) { return CloudSettings.ɵfac(t); } });
var ɵCloudSettings_BaseFactory = /*@__PURE__*/ ɵngcc0.ɵɵgetInheritedFactory(CloudSettings);
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(CloudSettings, [{
        type: Injectable
    }], null, null); })();
    return CloudSettings;
}(IonicNativePlugin));
export { CloudSettings };
