// @ionic-native/battery-status/__ivy_ngcc__/ngx/index.js
import { __decorate, __extends } from "tslib";
import { Injectable } from '@angular/core';
import { IonicNativePlugin, cordova } from '@ionic-native/core';
import { Observable } from 'rxjs';
import * as ɵngcc0 from '@angular/core';
var BatteryStatus = /** @class */ (function (_super) {
    __extends(BatteryStatus, _super);
    function BatteryStatus() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    BatteryStatus.prototype.onChange = function () { return cordova(this, "onChange", { "eventObservable": true, "event": "batterystatus" }, arguments); };
    BatteryStatus.prototype.onLow = function () { return cordova(this, "onLow", { "eventObservable": true, "event": "batterylow" }, arguments); };
    BatteryStatus.prototype.onCritical = function () { return cordova(this, "onCritical", { "eventObservable": true, "event": "batterycritical" }, arguments); };
    BatteryStatus.pluginName = "BatteryStatus";
    BatteryStatus.plugin = "cordova-plugin-battery-status";
    BatteryStatus.pluginRef = "navigator.battery";
    BatteryStatus.repo = "https://github.com/apache/cordova-plugin-battery-status";
    BatteryStatus.platforms = ["iOS", "Android", "Windows", "Browser"];
BatteryStatus.ɵfac = function BatteryStatus_Factory(t) { return ɵBatteryStatus_BaseFactory(t || BatteryStatus); };
BatteryStatus.ɵprov = ɵngcc0.ɵɵdefineInjectable({ token: BatteryStatus, factory: function (t) { return BatteryStatus.ɵfac(t); } });
var ɵBatteryStatus_BaseFactory = /*@__PURE__*/ ɵngcc0.ɵɵgetInheritedFactory(BatteryStatus);
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(BatteryStatus, [{
        type: Injectable
    }], null, null); })();
    return BatteryStatus;
}(IonicNativePlugin));
export { BatteryStatus };
