// @ionic-native/brightness/__ivy_ngcc__/ngx/index.js
import { __decorate, __extends } from "tslib";
import { Injectable } from '@angular/core';
import { IonicNativePlugin, cordova } from '@ionic-native/core';
import * as ɵngcc0 from '@angular/core';
var Brightness = /** @class */ (function (_super) {
    __extends(Brightness, _super);
    function Brightness() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    Brightness.prototype.setBrightness = function (value) { return cordova(this, "setBrightness", {}, arguments); };
    Brightness.prototype.getBrightness = function () { return cordova(this, "getBrightness", {}, arguments); };
    Brightness.prototype.setKeepScreenOn = function (value) { return cordova(this, "setKeepScreenOn", {}, arguments); };
    Brightness.pluginName = "Brightness";
    Brightness.plugin = "cordova-plugin-brightness";
    Brightness.pluginRef = "cordova.plugins.brightness";
    Brightness.repo = "https://github.com/mgcrea/cordova-plugin-brightness";
    Brightness.platforms = ["Android", "iOS"];
Brightness.ɵfac = function Brightness_Factory(t) { return ɵBrightness_BaseFactory(t || Brightness); };
Brightness.ɵprov = ɵngcc0.ɵɵdefineInjectable({ token: Brightness, factory: function (t) { return Brightness.ɵfac(t); } });
var ɵBrightness_BaseFactory = /*@__PURE__*/ ɵngcc0.ɵɵgetInheritedFactory(Brightness);
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(Brightness, [{
        type: Injectable
    }], null, null); })();
    return Brightness;
}(IonicNativePlugin));
export { Brightness };
